File: app/src/main/java/dmcs/astroWeather/activity/SettingsActivity.java
package dmcs.astroWeather.activity;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.os.Vibrator;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.util.List;

import dmcs.astroWeather.R;
import dmcs.astroWeather.db.DBLocalization;
import dmcs.astroWeather.db.DBParameter;
import dmcs.astroWeather.db.Localization;
import dmcs.astroWeather.util.Parameter;

/**
 * @author Mateusz Wieczorek
 */
public class SettingsActivity extends Activity {

    private DBLocalization db;
    private TextView city;
    private EditText refreshingValue;
    private Button saveButton;
    private Button selectButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.settings);
        db = new DBLocalization(this);
        init();
        initOnClicks();
    }

    @Override
    public void onBackPressed() {
        Intent intent = new Intent(SettingsActivity.this, MainActivity.class);
        startActivity(intent);
    }

    private void init() {
        city = (TextView) findViewById(R.id.settingsCity);
        refreshingValue = (EditText) findViewById(R.id.refreshingValue);
        saveButton = (Button) findViewById(R.id.saveButton);
        selectButton = (Button) findViewById(R.id.selectCity);
        setDefaultValues();
    }

    private void setDefaultValues() {
        city.setText(getResources().getString(R.string.settings_city) + ": " + String.valueOf(Parameter.LOCALIZATION_NAME));
        refreshingValue.setText(String.valueOf(Parameter.REFRESH_INTERVAL_IN_SEC));
    }

    private void initOnClicks() {
        saveButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Vibrator vb = (Vibrator) getSystemService(Context.VIBRATOR_SERVICE);

                try {
                    Parameter.REFRESH_INTERVAL_IN_SEC = Integer.valueOf(refreshingValue.getText().toString());
                    saveParametersToDatabase();
                    vb.vibrate(30);
                    Intent intent = new Intent(SettingsActivity.this, MainActivity.class);
                    startActivity(intent);
                } catch (NumberFormatException e) {
                    vb.vibrate(300);
                    Toast.makeText(SettingsActivity.this, "Nieprawidłowy format!", Toast.LENGTH_LONG).show();
                }
            }
        });

        selectButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Vibrator vb = (Vibrator) getSystemService(Context.VIBRATOR_SERVICE);
                vb.vibrate(30);
                Intent intent = new Intent(SettingsActivity.this, LocalizationListActivity.class);
                startActivity(intent);
            }
        });
    }

    private void saveParametersToDatabase() {
        DBParameter dbParameter = new DBParameter(this);
        dbParameter.updateParameter("LOCALIZATION_NAME", Parameter.LOCALIZATION_NAME);
        dbParameter.updateParameter("PRESSURE_UNIT", Parameter.PRESSURE_UNIT);
        dbParameter.updateParameter("REFRESH_INTERVAL_IN_SEC", String.valueOf(Parameter.REFRESH_INTERVAL_IN_SEC));
        dbParameter.updateParameter("TEMPERATURE_UNIT", Parameter.TEMPERATURE_UNIT);
        dbParameter.updateParameter("SPEED_UNIT", Parameter.SPEED_UNIT);
    }

}

File: app/src/main/java/com/weather/activity/SettingsActivity.kt
package com.weather.activity

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.os.Vibrator
import android.widget.Button
import android.widget.EditText
import android.widget.RadioButton
import android.widget.TextView

import com.weather.R
import com.weather.util.Parameter

/**
 * @author Mateusz Wieczorek
 */
class SettingsActivity : Activity() {

    private var refreshingValue: EditText? = null
    private var saveButton: Button? = null
    private var kilometerRadioButton: RadioButton? = null
    private var mileRadioButton: RadioButton? = null
    private var celsiusRadioButton: RadioButton? = null
    private var kelvinRadioButton: RadioButton? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings)
        init()
        initOnClicks()
    }

    override fun onBackPressed() {
        val intent = Intent(this@SettingsActivity, MainActivity::class.java)
        startActivity(intent)
    }

    private fun init() {
        refreshingValue = findViewById(R.id.refreshingValue)
        saveButton = findViewById(R.id.saveButton)
        celsiusRadioButton = findViewById(R.id.celsiusRadioButton)
        kelvinRadioButton = findViewById(R.id.kelvinRadioButton)
        kilometerRadioButton = findViewById(R.id.kilometerRadioButton)
        mileRadioButton = findViewById(R.id.mileRadioButton)

        setTemperatureUnits()
        setSpeedUnit()

        val city = findViewById<TextView>(R.id.settingsCity)
        city.text = resources.getString(R.string.settings_city) + ": " + Parameter.LOCALIZATION_NAME.toString()
        refreshingValue!!.setText(Parameter.REFRESH_INTERVAL_IN_SEC.toString())
    }

    private fun initOnClicks() {
        saveButton!!.setOnClickListener { v ->
            val vb = getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
            Parameter.REFRESH_INTERVAL_IN_SEC = Integer.valueOf(refreshingValue!!.text.toString())!!
            val intent = Intent(this@SettingsActivity, MainActivity::class.java)
            startActivity(intent)
        }

        kilometerRadioButton!!.setOnClickListener { v -> val vb = getSystemService(Context.VIBRATOR_SERVICE) as Vibrator }

        mileRadioButton!!.setOnClickListener { v -> val vb = getSystemService(Context.VIBRATOR_SERVICE) as Vibrator }

        celsiusRadioButton!!.setOnClickListener { v -> val vb = getSystemService(Context.VIBRATOR_SERVICE) as Vibrator }

        kelvinRadioButton!!.setOnClickListener { v -> val vb = getSystemService(Context.VIBRATOR_SERVICE) as Vibrator }
    }

    private fun setSpeedUnit() {
        if (Parameter.SPEED_UNIT == "km/h") {
            kilometerRadioButton!!.isChecked = true
            mileRadioButton!!.isChecked = false
        } else {
            kilometerRadioButton!!.isChecked = false
            mileRadioButton!!.isChecked = true
        }
    }

    private fun setTemperatureUnits() {
        if (Parameter.TEMPERATURE_UNIT == "°C") {
            celsiusRadioButton!!.isChecked = true
            kelvinRadioButton!!.isChecked = false
        } else {
            celsiusRadioButton!!.isChecked = false
            kelvinRadioButton!!.isChecked = true
        }
    }

}
